core/src/tools/testfind_fd.cc

```diff
@@ -0,0 +1,287 @@
+/*
+   BAREOS® - Backup Archiving REcovery Open Sourced
+
+   Copyright (C) 2022-2023 Bareos GmbH & Co. KG
+
+   This program is Free Software; you can redistribute it and/or
+   modify it under the terms of version three of the GNU Affero General Public
+   License as published by the Free Software Foundation and included
+   in the file LICENSE.
+
+   This program is distributed in the hope that it will be useful, but
+   WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+   Affero General Public License for more details.
+
+   You should have received a copy of the GNU Affero General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+   02110-1301, USA.
+*/
+
+#include "dird/fd_sendfileset.h"
+#include "filed/filed_utils.h"
+#include "filed/filed.h"
+#include "filed/filed_jcr_impl.h"
+#include "filed/filed_globals.h"
+#include "filed/dir_cmd.h"
+#include "filed/fileset.h"
+#include "findlib/attribs.h"
+#include "lib/parse_conf.h"
+#include "tools/dummysockets.h"
+#include "tools/testfind_fd.h"
+
+#include <iostream>
+
+using namespace filedaemon;
+
+static struct TestfindStats {
+  uint64_t num_files = 0;
+  uint64_t max_file_len = 0;
+  uint64_t max_path_len = 0;
+  uint64_t trunc_fname = 0;
+  uint64_t trunc_path = 0;
+  uint64_t hard_links = 0;
+  bool print_attributes = false;
+} testfind_stats;
+
+
+static int handleFile(JobControlRecord* jcr,
+                      FindFilesPacket* ff_pkt,
+                      bool top_level)
+{
+  if (!SaveFile(jcr, ff_pkt, top_level)) { return 0; }
+  if (!PrintFile(jcr, ff_pkt, top_level)) { return 0; }
+  return 1;
+}
```

Comment:
the whole purpose of this PR is to make testfind simulate what FD does. The FD uses the `SaveFile(..)` callback to do all the reading, encryption, etc ... 
Without the `SaveFile(..)` we would be missing the point of the "faking", and we would not be able to find errors (in case there are) when manipulating the files
`PrintFile(...)` is only there to print what was succesfully "saved". Using it alone is not enough

I think I should add some kind of test with some faulty files or something like that to make sure that problems are detected.